examples/src/bin/custom_editable.rs

```diff
@@ -0,0 +1,140 @@
+//! # Custom GdkPaintable
+//!
+//! This example shows how to create a custom gdk::Paintable
+//! that renders a square with a linear color gradient
+
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::{gdk, glib, graphene, gsk};
+use std::env::args;
+
+mod imp {
+    use super::*;
+
+    pub struct CustomEditable {
+        pub text: gtk::Text,
+    }
+
+    impl Default for CustomEditable {
+        fn default() -> Self {
+            Self {
+                text: gtk::Text::new(),
+            }
+        }
+    }
+
+    #[glib::object_subclass]
+    impl ObjectSubclass for CustomEditable {
+        const NAME: &'static str = "CustomEditable";
+        type Type = super::CustomEditable;
+        type ParentType = gtk::Widget;
+        type Interfaces = (gtk::Editable,);
+
+        fn class_init(klass: &mut Self::Class) {
+            let total_props = klass.install_properties();
```

Comment:
This could be automatically called as part of `<Editable as IsImplementable>::class_init()`

Reply:
Can it actually be done at that stage? we only have `interface_init` and I don't have any idea how to cast `iface` to a `ClassStruct` or just a `GObjectClass`

Reply:
Get the class struct for the type implementing the interface: https://gtk-rs.org/gtk-rs/glib/object/struct.Interface.html#method.get_instance_type and https://gtk-rs.org/gtk-rs/glib/object/struct.Class.html#method.from_type